### PR TITLE
cleanWal by commitLogId

### PR DESCRIPTION
**Pull request by liwenhui-soul**
## What type of PR is this?
- [x] bug
- [ ] feature
- [ ] enhancement

## Checklist:
Tests:
- [ ] Unit test(positive and negative cases)
- [ ] Function test
- [ ] Performance test
- [ ] N/A

Affects:
- [ ] Documentation affected (Please add the label if documentation needs to be modified.)
- [ ] Incompatibility (If it breaks the compatibility, please describe it and add the label.）
- [ ] If it's needed to cherry-pick (If cherry-pick to some branches is required, please label the destination version(s).)
- [ ] Performance impacted: Consumes more CPU/Memory

## Release notes:

Please confirm whether to be reflected in release notes and how to describe:
> ex. Fixed the bug .....

### CODE REVIEW

File: src/kvstore/NebulaStore.cpp [hunk 1]
```diff
@@ -1133,7 +1133,7 @@ void NebulaStore::cleanWAL() {
       auto& part = partEntry.second;
       if (part->needToCleanWal()) {
         // clean wal by expired time
-        part->wal()->cleanWAL();
+        part->cleanWal();
```
Comment (critical27) [inline src/kvstore/NebulaStore.cpp:1136]: Maybe unify all usage, either implement `cleanWAL()` in `Part` and `Listener`, or just call `part->wal()->cleanWAL(id)`.

**Review by critical27 — COMMENTED** [review]
Generally LGTM

File: src/kvstore/NebulaStore.cpp [hunk 2]
```diff
@@ -1133,16 +1133,16 @@ void NebulaStore::cleanWAL() {
...
     }
   }
   for (const auto& spaceEntry : spaceListeners_) {
     for (const auto& partEntry : spaceEntry.second->listeners_) {
       for (const auto& typeEntry : partEntry.second) {
         const auto& listener = typeEntry.second;
-        // clean wal by log id
-        listener->wal()->cleanWAL(listener->getApplyId());
+        // clean wal by commit log id
+        listener->cleanWal();
```
Comment (critical27) [inline src/kvstore/NebulaStore.cpp:1145]: `Listener` need to override it, because it use apply id, not commitId

**Review by critical27 — APPROVED** [review]
Good job

File: src/kvstore/wal/FileBasedWal.cpp [hunk 3]
```diff
@@ -672,23 +672,30 @@ void FileBasedWal::cleanWAL() {
...
             << walFiles_.rbegin()->second->lastId();
     return;
   }
-
+  int walTTL = FLAGS_wal_ttl;
   // remove wal file that lastId is less than target
   auto iter = walFiles_.begin();
   while (iter != walFiles_.end()) {
-    if (iter->second->lastId() < id) {
+    if (iter->second->lastId() < id && index < size - 2 && (now - iter->second->mtime() > walTTL)) {
```
Comment (pengweisong) [inline src/kvstore/wal/FileBasedWal.cpp:694]: If we do not use walTTL to limit the cleanWAL, what will happen?

Comment (critical27) [reply src/kvstore/wal/FileBasedWal.cpp:694]: Then no wal would be deleted

**Review by pengweisong — APPROVED** [review]
Good Job

**Pull request merged**